opengovsg/memos
# feat: create/read templates

## Context

C+R for templates.

## Approach

Adds these endpoints with their accompanying logic: 

- [x] `POST /api/templates`
- [x] `GET /api/templates/:id`

**TODOs for the future**:

- Add tests and figure out Nest's dependency injection when running tests
- Improve API docs

## Tests

Manual tests for now 🥲 

- `POST /api/templates` - Send a request, check that the necessary records are inserted (template, template version, editor and issuer) 
- `GET /api/templates/:id` - Send a request, check that the latest template version is returned
